File: reni-backend/src/main/java/pe/gob/mimp/reni/dto/response/PerfilFuncionalidadListarResponse.java
package pe.gob.mimp.reni.dto.response;

import java.io.Serializable;
import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class PerfilFuncionalidadListarResponse implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long idFuncionalidad;
	private Long idFuncionalidadPadre;
	private String titulo;
	private String referencia;
	private List<PerfilFuncionalidadListarResponse> listaFuncionalidad;
}

File: reni-frontend/src/main/reni-frontend/src/app/modules/intranet/dto/response/perfil-funcionalidad-listar.response.ts
export class PerfilFuncionalidadListarResponse {
    idFuncionalidad: number;
    idFuncionalidadPadre: number;
    titulo: string;
    referencia: string;
    listaFuncionalidad: PerfilFuncionalidadListarResponse[];

    selected: boolean = false;
    existe: boolean = false;
}